solidity/random-beacon/contracts/libraries/Groups.sol

```diff
@@ -14,17 +14,28 @@ library Groups {
 
     struct Group {
         bytes groupPubKey;
-        uint256 activationTimestamp;
+        uint256 activationBlockNumber;
         uint32[] members;
+        // When selected group does not create a relay entry on-time it should
+        // be marked as terminated.
+        bool terminated;
     }
 
     struct Data {
         // Mapping of keccak256 hashes of group public keys to groups details.
         mapping(bytes32 => Group) groupsData;
         // Holds keccak256 hashes of group public keys in the order of registration.
         bytes32[] groupsRegistry;
+        // Group ids that were active but failed creating a relay entry.
+        uint64[] activeTerminatedGroups;
         // TODO: Remember about decreasing the counter in case of expiration or termination.
         uint64 activeGroupsCount;
+        // Points to the first active group, it is also the expired groups counter.
+        uint64 expiredGroupOffset;
+        // Lifetime of a group in blocks after which group expires.
+        uint256 groupLifetime;
```

Comment:
`RandomBeacon` defines `groupLifetime` as `@notice Group lifetime in seconds`, we need to make sure it's consistent.

Reply:
Sure, done as part of fd01258